app/domain/authentication/validate_authenticator_exist.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module Authentication
+
+  ValidateAuthenticatorExist = CommandClass.new(
+    dependencies: {
+    },
+    inputs: %i(input authenticator)
+  ) do
+
+    def call
+      validate_authenticator_exists
+    end
+
+    private
+
+    def validate_authenticator_exists
+      raise AuthenticatorNotFound, @input.authenticator_name unless @authenticator
```

Comment:
As we discussed the other day, I believe the class `ValidateAuthenticatorExists` is superfluous.  Where you would write in client code:
```
@validate_authenticator_exists.(input, authenticator)
```
you can just as easily write:
```
raise AuthenticatorNotFound, input.authenticator_name unless authenticator
```
Note that in general a CommandClass with no dependencies is equivalent to a pure function whose arguments are the same arguments as the command class's `call` method.  

In this case, it's not technically pure because you can raise an error.  However, you might as well just use a function with a verb phrase name or, if the function is only a single line as is the case here, you can inline it where you need it.  But I think adding the machinery of `CommandClass` here is unnecessary and potentially confusing.

Reply:
Yes, I was doubtful about this one